service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -243,4 +255,20 @@ private String buildPriorPrescription(MedicationRequest medicationRequest) {
         }
         return StringUtils.EMPTY;
     }
+
+    private String buildParticipant(MedicationRequest medicationRequest) {
+        Predicate<String> isPractitioner = s -> s.startsWith(ResourceType.Practitioner.name());
+        Predicate<String> isPractitionerRole = s -> s.startsWith(ResourceType.PractitionerRole.name());
+        Predicate<String> isOrganization = s -> s.startsWith(ResourceType.Organization.name());
+        Predicate<String> isRelevant = isPractitioner.or(isPractitionerRole).or(isOrganization);
+
+        if (medicationRequest.hasRecorder() && medicationRequest.getRecorder().hasReference()) {
+            final var reference = medicationRequest.getRecorder();
+            if (isRelevant.test(reference.getReference())) {
+                return participantMapper.mapToParticipant(
+                    messageContext.getIdMapper().get(reference), ParticipantType.AUTHOR);
+            }
+        }
+        return StringUtils.EMPTY;
```

Comment:
Only Practitioner and and Patient are profiled, so recorder cannot reference a PractitionerRole or organisation here.

So anything other than Practitioner or Patient we should raise an error.

For patient: 

> if   recorder is Patient  then prepend   string "Recorded By Patient " to pertinentInformation text

Reply:
Is the AC wrong?

> WHEN MedicationRequest.Recorder has practitioner/practitonerRole/Organisation reference

Or the others just need to be dealt with another way?

Reply:
Yes it looks like the AC is wrong in this case. The mapping spec is always the "source of truth" of they don't agree

Reply:
Which pertinentInformation does that refer to @mattd-kainos ?

There's:
`MedicationStatement.pertinentInformation.pertinentMedicationDosage`
`MedicationStatement.component.ehrSupplyAuthorise.pertinentInformation.pertinentSupplyAnnotation`
and the spreadsheet also describes this for `note`:
```
<pertinentInformation typeCode="PERT">
  <sequenceNumber value="+1"/>
  <pertinentAnnotation classCode="OBS" moodCode="EVN">
    <text>{note}</text>
  </pertinentAnnotation>
</pertinentInformation>
```
but which is not yet in the relevant mustache template

Reply:
It seems I was looking at allergy instead of medication when I wrote this. MedicationRequest:G36 is the right place to look

recorder | I | 1..1 | Reference ( CareConnect-GPC-Practitioner-1 ) | Person who entered the request | y | If Reference(Practitioner)   populated generate an agentPerson Agent corresponding to the   Practitioner/PractitionerRole/Organisation and reference as a Participant of   type AUT at MedicationStatement level - not ehrSupplyX level     <Participant typeCode="AUT"   contextControlCode="OP"><agentRef   classCode="AGNT"><id root="Uuid for   participant"/></agentRef></Participant>
-- | -- | -- | -- | -- | -- | --